Yosemite/YosemiteTests/Stores/CustomerStoreTests.swift

```diff
@@ -0,0 +1,106 @@
+import XCTest
+@testable import Networking
+@testable import Yosemite
+
+class CustomerStoreTests: XCTestCase {
+
+    private var dispatcher: Dispatcher!
+    private var storageManager: MockStorageManager!
+    private var network: MockNetwork!
+    private var remote: CustomerRemote!
+    private var store: CustomerStore!
+    private let dummySiteID: Int64 = 12345
+    private let dummyCustomerID: Int64 = 1
+
+    override func setUp() {
+        super.setUp()
+        dispatcher = Dispatcher()
+        storageManager = MockStorageManager()
+        network = MockNetwork()
+        remote = CustomerRemote(network: network)
+        store = CustomerStore(
+            dispatcher: dispatcher,
+            storageManager: storageManager,
+            network: network,
+            remote: remote
+        )
+    }
+
+    override func tearDown() {
+        dispatcher = nil
+        storageManager = nil
+        network = nil
+        remote = nil
+        store = nil
+        super.tearDown()
+    }
+
+    func test_retrieveCustomer_returns_Customer_upon_success() {
+        // Given
+        network.simulateResponse(requestUrlSuffix: "customers/\(dummyCustomerID)", filename: "customer")
+
+        // When
+        let result: Result<Customer, Error> = waitFor { promise in
+            let action = CustomerAction.retrieveCustomer(siteID: self.dummySiteID, customerID: self.dummyCustomerID) { result in
+                promise(result)
+            }
+            self.store.onAction(action)
+        }
+
+        // Then
+        XCTAssertTrue(result.isSuccess)
+    }
+
+    func test_retrieveCustomer_returns_notFound_Error_upon_siteID_error() {
+        // Given
+        let expectedError = NetworkError.notFound
+        network.simulateError(requestUrlSuffix: "customers/\(dummyCustomerID)", error: expectedError)
+
+        // When
+        let result: Result<Customer, Error> = waitFor { promise in
+            let action = CustomerAction.retrieveCustomer(siteID: 999, customerID: self.dummyCustomerID) { result in
```

Comment:
Does this actually test what it says it tests? Reading it, it looks like the same result would be given whatever the `siteID` was, including if it were `dummySiteID`

Reply:
You're right, it doesn't, no. It would give the same result no matter the siteID because we're calling `simulateError()` in the MockNetwork 🤔 

From our last session, knowing that we should test the API/responsibilities of the class, and that we're implicitly testing both Remote and Mapper as well, I went ahead for now and simplified the [CustomerStoreTests](https://github.com/woocommerce/woocommerce-ios/blob/e7cbe306e70cbf8861a1743b7eb0eb847e0e9fca/Yosemite/YosemiteTests/Stores/CustomerStoreTests.swift) to 2 cases only: 
- When attempting to retrieve a Customer, does it return a Customer upon success? 
- When attempting to retrieve a Customer, does it return an Error upon failure? 

At this point, and without a Mock, the responsibility of checking the siteID and the customerID should be tested on `CustomerRemoteTests` rather than these for the `CustomerStoreTests`, as whatever we pass within those parameters will just return the same result due to using the `MockNetwork`. What do you think?